src/app.js

```diff
@@ -30,53 +30,54 @@ client.on('ready', () => {
 });
 
 // Create an event listener for messages
-client.on('message', msg => {
-  const { content, author, channel, member } = msg;
+client.on('message', async msg => {
+  // Ignore messages from the bot
+  if (msg.author.bot) return;
 
-  // Returns true if the message content matches any of
-  // the given commands
-  const commandMatches = commands => {
-    if (Array.isArray(commands)) {
-      for (let i = 0; i < commands.length; i++) {
-        if (`${prefix}${commands[i]}` === content.split(' ')[0]) return true;
-      }
-      return false;
-    }
+  //Ignore any message that does not start with prefix
+  if (msg.content.indexOf(prefix) !== 0) return;
 
-    return `${prefix}${commands}` === content;
-  };
+  //   console.log(`content: ${msg.content}
+  // author: ${msg.author}
+  // channel: ${msg.channel}
+  // member: ${msg.member}
+  // `);
 
-  // Do not evaluate a message not starting with  our prefix
-  if (content.startsWith(prefix)) {
-    switch (true) {
-      case commandMatches('ping'):
-        channel.send('pong!');
-        break;
-      case commandMatches('knowme') && !author.bot:
-        knowme(msg);
-        break;
-      case commandMatches('repopulate') &&
-        member.roles.has('363153451833753600'):
-        repopulate(msg);
-        break;
-      case commandMatches(['marshmallow', 'mm']):
-        marshmallow(msg);
-        break;
-      case commandMatches('mine'):
-        mine(msg);
-        break;
-      case content.match(/^(!help|!h) !?[a-zA-Z]+$/) !== null:
-        helpSpecific(msg);
-        break;
-      case commandMatches(['help', 'h']):
-        help(msg);
-        break;
-      case commandMatches(['emoji']):
-        emoji(msg);
-        break;
-      default:
-        break;
-    }
+  const args = msg.content
+    .slice(prefix.length)
+    .trim()
+    .split(/ +/g);
+  const command = args.shift().toLowerCase();
+
+  switch (command) {
+    case 'ping':
+      msg.channel.send('pong!');
+      break;
+    case 'knowme':
+      knowme(msg);
+      break;
+    case 'repopulate':
+      if (msg.member.roles.has('363153451833753600')) repopulate(msg);
+      break;
+    case 'marshmallow':
+    case 'mm':
+      marshmallow(msg);
+      break;
+    case 'mine':
+      mine(msg);
+      break;
+    case 'help':
+    case 'h':
+      if (args.length === 0) help(msg);
+      else if (args[0].match(new RegExp(`^${prefix}?\\w+`, 'g')))
```

Comment:
You have an extra `\` there, right now it's just looking for the `w` character. Also, I'm not sure why you're going through this step instead of just doing `helpSpecific(msg, args[0].slice(prefix.length).trim())`

Reply:
1. About \w:  
    **/\w+/**  equals  **new RegExp('\\w+')** [link](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp)
2. Why:
    if the **content** string will be like **!h !mine** all will work, but if it will be **!h +mine** or **!h mine**? Behavior may differ from the expected.

Reply:
1. `//w` = `w`: https://regexr.com/3rqmq
2. `!help` should not accept anything other than `!mine`. It should error out for anything else, wheter it be `mine` or `+mine` or `mine a bced`